compiler/luci-interpreter/src/kernels/RoPE.cpp

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (c) 2024 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "kernels/RoPE.h"
+
+#include "kernels/Utils.h"
+
+namespace luci_interpreter
+{
+namespace kernels
+{
+
+RoPE::RoPE(const Tensor *input, const Tensor *sin_table, const Tensor *cos_table,
+          Tensor *output) : Kernel({input, sin_table, cos_table}, {output})
+{
+}
+
+void RoPE::configure()
+{
+  LUCI_INTERPRETER_CHECK(input()->shape().num_dims() == 4);
+
+  output()->resize(input()->shape());
+}
+
+void RoPE::execute() const
+{
+  switch (input()->element_type())
+  {
+    case DataType::FLOAT32:
+      evalFloat();
+      break;
+    default:
+      throw std::runtime_error("luci-rope Unsupported type.");
+  }
+}
+
+void RoPE::evalFloat() const
+{
+  const auto input_shape = getTensorShape(input());
+  const auto sin_table_shape = getTensorShape(sin_table());
+  const auto cos_table_shape = getTensorShape(cos_table());
+  auto output_shape = getTensorShape(output());
+
+  const float *input_data = getTensorData<float>(input());
+  const float *sin_table_data = getTensorData<float>(sin_table());
+  const float *cos_table_data = getTensorData<float>(cos_table());
+  float *output_data = getTensorData<float>(output());
+
+  if (input_shape.DimensionsCount() == 4)
+  {
+    const int32_t i0_n = input_shape.Dims(0);
+    const int32_t multihead_n = input_shape.Dims(1);
+    const int32_t i2_n = input_shape.Dims(2);
+    const int32_t head_n = input_shape.Dims(3);
```

Comment:
i think the meaning of the variable name is ambiguous.
the meaning of i0 and i2 is input rank0 and input rank2.
I would like to change this to a more clear name, but since I didn't exactly understand the meaning of this rank,
so I used a simple name(ex. i0, i2.).
If recommend variable name with a clear meaning, I will change it.
and,
as a kernel added to verify fused ops, optimization is not considered and is implemented as is in the graph.
and, although it may already be optimized,
there is no environment (input, golden data?) for measuring latency yet (maybe i don't know),
so i don't know what environment the data should be measured to be meaningful. andi plan to further

Reply:
> rank0 and input rank2.

usually we used term `dim` as `dimension` for this.
meaning for each dim, maybe the source paper can explain or the python script for the model.

> for measuring latency yet

we don't measure this in frontend :)

> i don't know what environment the data should be measured to be meaningful.

as of now, we don't. just random input data, compared to source model, onnx-runtime for onnx and tflite interpreter for tflite model, looked as some computational black box.